public/app/features/dashboard-scene/scene/PanelRepeaterGridItem.tsx

```diff
@@ -0,0 +1,248 @@
+  });
+
+  private _ignoreNextStateChange = false;
+  private _isWaitingForVariables = false;
+
+  public constructor(state: PanelRepeaterGridItemState) {
+    super(state);
+
+    this.addActivationHandler(() => this._activationHandler());
+  }
+
+  private _activationHandler() {
+    this._subs.add(this.subscribeToState((newState, prevState) => this._handleGridResize(newState, prevState)));
+
+    // If we our variable is ready we can process repeats on activation
+    if (sceneGraph.hasVariableDependencyInLoadingState(this)) {
+      this._isWaitingForVariables = true;
+    } else {
+      this._performRepeat();
+    }
+  }
+
+  private _onVariableChanged(changedVariables: Set<SceneVariable>, dependencyChanged: boolean): void {
+    if (dependencyChanged) {
+      this._performRepeat();
+      return;
+    }
+
+    // If we are waiting for variables and the variable is no longer loading then we are ready to repeat as well
+    if (this._isWaitingForVariables && !sceneGraph.hasVariableDependencyInLoadingState(this)) {
+      this._isWaitingForVariables = false;
+      this._performRepeat();
+    }
+  }
+
+  /**
+   * Uses the current repeat item count to calculate the user intended desired itemHeight
+   */
+  private _handleGridResize(newState: PanelRepeaterGridItemState, prevState: PanelRepeaterGridItemState) {
+    if (this._ignoreNextStateChange) {
+      this._ignoreNextStateChange = false;
+      return;
+    }
+
+    const itemCount = this.state.repeatedPanels?.length ?? 1;
+    const stateChange: Partial<PanelRepeaterGridItemState> = {};
+
+    // Height changed
+    if (newState.height !== prevState.height) {
+      if (this.getRepeatDirection() === 'v') {
+        const itemHeight = Math.ceil(newState.height! / itemCount);
+        stateChange.itemHeight = itemHeight;
+      } else {
+        const rowCount = Math.ceil(itemCount / this.getMaxPerRow());
+        stateChange.itemHeight = Math.ceil(newState.height! / rowCount);
+      }
+    }
+
+    this._ignoreNextStateChange = true;
+    this.setState(stateChange);
+  }
+
+  private _performRepeat() {
+    const variable = sceneGraph.lookupVariable(this.state.variableName, this);
+    if (!variable) {
+      console.error('SceneGridItemRepeater: Variable not found');
+      return;
+    }
+
+    if (!(variable instanceof MultiValueVariable)) {
+      console.error('PanelRepeaterGridItem: Variable is not a MultiValueVariable');
+      return;
+    }
+
+    const panelToRepeat = this.state.source;
+    const { values, texts } = this.getVariableValues(variable);
+    const repeatedPanels: VizPanel[] = [];
+
+    // Loop through variable values and create repeates
+    for (let index = 0; index < values.length; index++) {
+      const clone = panelToRepeat.clone({
+        $variables: new SceneVariableSet({
+          variables: [
+            new ConstantVariable({ name: variable.state.name, value: values[index], text: String(texts[index]) }),
```

Comment:
This will be changed to `LocalValueVariable` I assume?

Reply:
yes, just pushed that update